python/lsst/daf/butler/core/storageClass.py

```diff
@@ -271,3 +301,18 @@ def registerStorageClass(self, storageClass):
                                  " definition")
         else:
             self._storageClasses[storageClass.name] = storageClass
+
+    def unregisterStorageClass(self, storageClassName):
+        """Remove the named StorageClass from the factory.
+
+        Parameters
+        ----------
+        storageClassName : `str`
+            Name of storage class to remove.
+
+        Raises
+        ------
+        KeyError
+            The named storage class is not registered.
+        """
+        del self._storageClasses[storageClassName]
```

Comment:
This seems to go against the principle that `StorageClassFactory` is append-only.

Reply:
Yes, but for *testing* purposes it's incredibly useful...

Reply:
Sounds like we should just document that intent, then.